src/track/keyutils.cpp

```diff
@@ -16,18 +16,23 @@ using mixxx::track::io::key::ChromaticKey;
 using mixxx::track::io::key::ChromaticKey_IsValid;
 
 // OpenKey notation, the numbers 1-12 followed by d (dur, major) or m (moll, minor).
-static const QString s_openKeyPattern("^\\s*(1[0-2]|[1-9])([dm])\\s*$");
+static const QRegularExpression s_openKeyRegex(QStringLiteral(
```

Comment:
Why are these regexes now enclosed in `\\A(?:...)\\z`? Please leave a comment, this is not obvious to me.

Reply:
It is to replicate the old `exactMatch` API of QRegEx: https://doc-snapshots.qt.io/qt6-dev/qtcore-changes-qt6.html#exactmatch-vs-match-hasmatch

Reply:
QRegularExpression::anchoredPattern was introduced in Qt 5.15 but we need to support Qt 5.12 for macOS. So I mimicked the implementation of QRegularExpression::anchoredPattern.

Reply:
comment added